src/wfs/src/main/java/org/geoserver/wfs/xml/RestrictionToXSDConstrainingFacetVisitor.java

```diff
@@ -0,0 +1,75 @@
+/* (c) 2025 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wfs.xml;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.eclipse.xsd.XSDConstrainingFacet;
+import org.eclipse.xsd.XSDEnumerationFacet;
+import org.eclipse.xsd.XSDFactory;
+import org.eclipse.xsd.XSDMaxInclusiveFacet;
+import org.eclipse.xsd.XSDMinInclusiveFacet;
+import org.geotools.api.filter.And;
+import org.geotools.api.filter.Filter;
+import org.geotools.api.filter.Not;
+import org.geotools.api.filter.Or;
+import org.geotools.api.filter.PropertyIsBetween;
+import org.geotools.api.filter.PropertyIsEqualTo;
+import org.geotools.filter.visitor.AbstractFilterVisitor;
+
+@SuppressWarnings("unchecked")
+public class RestrictionToXSDConstrainingFacetVisitor extends AbstractFilterVisitor {
```

Comment:
Maybe add a comment that this visitor is expected to work only with the well known flter patterns (explain which ones) and won't to do right thing if a manual validation filter is added, e.g.. "att < 10 or att > 20 and not (add = 15 or add = 30)" will not be translated correctly

Reply:
Done. Thanks!